frontend/src/App.tsx

```diff
@@ -916,9 +925,37 @@ export class App extends PureComponent<Props, State> {
       queryString = queryString.substring(1)
     }
 
+    const baseUriParts =
+      this.connectionManager && this.connectionManager.getBaseUriParts()
+
+    if (!baseUriParts) {
+      logError("Cannot send rerun backmessage when disconnected from server.")
+      return
+    }
+
+    const { basePath } = baseUriParts
+
+    // NOTE: We specifically check for `null` or `undefined` instead of making
+    //       a falsy check below because navigating to "" can be used to
+    //       navigate to the main page of an app.
+    if (pageName === undefined) {
+      // If pageName is undefined, we're not switching pages, so we should
+      // use the current URL to determine the pageName.
+      pageName = document.location.pathname.replace(
+        new RegExp(`^/${basePath}/?`),
```

Comment:
`basePath` theoretically could contain certain special characters - `.`, `*`, `$` - which I assume are not automatically escaped? ... I think this might cause problems here since it might be interpreted as regex instructions. Maybe a non-regex replace might be a safer option here?

Reply:
Ah, that's true.

I think it'll be necessary to do two replaces here:
* the first to remove `^/${basePath}`
* the second to (possibly) remove a leading `/` that may still remain, which should be fine to write as a regex.

It's less clean than a single regex but is likely necessary due to basePath potentially containing unescaped characters. I'll slightly rework things to do this and add a comment explaining why